### ISSUE TITLE
Read/Write method of DataBuffer is against intuition

### ISSUE THREAD

**Issue by kevinhuangwl** [tensorflow/java#510]
Naturally when using DataBuffer.read(sth), should expect it to read from sth into the buffer, and vice versa.

But the API of DataBuffer is now just the opposite

**Comment by karllessard** [comment]
Yeah, I tend to agree with you... if I recall correctly, the reasoning behind this was to match the API of the standard NIO buffer, where the [`get`](https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#get-byte:A-) method (so a "read" operation) writes to a destination buffer, while the `put` method (write operation) reads from a source.

I think the confusion in NdArray is that it is not clear to which subject is attached the action read/write. How you describe it, the subject is the buffer itself, while in reality it is the caller who is taking the action to read or write to/from the buffer.

Renaming them to `readTo` and `writeFrom` would make that clearer and would have less impact to existing users than simply switching their name.
